stan/math/fwd/core/fvar.hpp

```diff
@@ -20,29 +19,63 @@ namespace stan {
 
       typedef fvar value_type;
 
-      fvar() : val_(0.0), d_(0.0) { }
-
+      /**
+       * Construct a forward variable with zero value and zero
+       * tangent.
+       */
+      fvar() : val_(0), d_(0) { }
+
+      /**
+       * Copy construct a forward variable by copying its value and
+       * tangent.
+       *
+       * @param x forward variable to copy
+       */
       fvar(const fvar<T>& x) : val_(x.val_), d_(x.d_) {  }
 
-      fvar(const T& v) : val_(v), d_(0.0) {  // NOLINT(runtime/explicit)
-        if (is_nan(v))
+      /**
+       * Construct a forward variable with specified value and zero
+       * tangent.  If the value is not-a-number, tangent will be set to
+       * not-a-number.
+       *
+       * @param v value of constructed forward variable
+       */
+      fvar(double v) : val_(v), d_(0) {  // NOLINT
```

Comment:
Is it okay that this constructor used to take any type `T` and now just takes `double`? Is it because of the other constructors with default values for the derivative?